### ISSUE TITLE
Watcher ignoring files in hidden folders

### ISSUE THREAD

**Issue by schmod** [BrowserSync/browser-sync#845]
Changed-file detection is currently broken if a user requests to watch files in a hidden directory (eg. `.tmp`)

In my experience, it's pretty common for web developers to use a `.tmp` directory for generated sources (ie. the CSS output of SASS/LESS).  Among other things, many Yeoman generators follow this convention.

This regression appears to have been introduced by 33d49d29dc668474d9dd42d4cb174a3213a37979 (https://github.com/BrowserSync/grunt-browser-sync/issues/118).  

---

What's the rationale for ignoring dotfiles by default?  

If we want to document this behavior, I guess I'm fine with that, although it seems extremely counterintuitive for BrowserSync to ignore a path that I specifically asked for in my glob pattern.

Here's an sample grunt config that demonstrates how I use `.tmp` with BrowserSync:

```js
{
    browserSync: {
        bsFiles: {
            src: [
                'app/*.html',
                'app/scripts/*.js',
                '.tmp/styles/*.css'
            ]
        },
        options: {
            watchTask: true,
            server: {
                baseDir: ['.tmp', './app']
            }
        }
    }
}
```

**Comment by Plou** [comment]
This refer to https://github.com/BrowserSync/browser-sync/pull/776

tl;dr :
The old ignored path came from a regex from Chokidar's readme which purpose was to ignore dotfiles. I changed it so it would no prevent us to go up the tree.

As @shakyShane said https://github.com/BrowserSync/browser-sync/pull/776#issuecomment-138905100, It could be a list of commonly used folders and files.

**Comment by shakyShane** [comment]
I think the best solution here is to simply not ignore anything by default.

**Comment by schmod** [comment]
Well, that was quick.  Thanks!  :-D

**Comment by shakyShane** [comment]
browser-sync@2.9.7

**Comment by shakyShane** [comment]
and you're welcome.
